### PR TITLE
Check that QR file paths are readable and are not directories

### PR DESCRIPTION
**Pull request by oz123**
This is a possible fix for https://github.com/kairos-io/kairos/issues/568.

### CODE REVIEW

File: internal/cli/register.go [hunk 1]
```diff
@@ -4,16 +4,50 @@ import (
...
 	"io/ioutil"
+	"os"
 
 	nodepair "github.com/mudler/go-nodepair"
 	qr "github.com/mudler/go-nodepair/qrcode"
 )
 
+// isDirectory determines if a file represented
+// by `path` is a directory or not
+func isDirectory(path string) (bool, error) {
```
Comment (mauromorales) [inline internal/cli/register.go:15]: these two functions look quite generic, might be worth putting on a different package to be used by multiple commands from the CLI ... maybe even shared across project, not sure what't the architectural goal there

**Comment by mauromorales** [comment]
:sweat_smile: Sorry for putting this comment in the PR, I'm still figuring out what the right channels of communication are.

Do we have clear guidelines on testing on the different projects? I'm right now hesitating to ask whether tests should be added for this bug fix, and what kind of tests.

Also, does CI check for a min % of coverage? Not trying to impose a way of working, just trying to familiarize myself and make mental notes on which suggestions to make on the different projects.

Thank you all 

cc @oz123 @mudler @jimmykarily

**Comment by oz123** [comment]
These are good points. I've been asking myself the same questions as I'm also new to the project.

**Review by mauromorales — APPROVED** [review]
lgtm

**Pull request merged**